### PR TITLE
bpo-42562: Fix issue when dis failed to parse function that has no line numbers

### PR DESCRIPTION
**Pull request by uriyyo**
https://bugs.python.org/issue42562

### CODE REVIEW

**Comment by markshannon** [comment]
https://github.com/python/cpython/pull/23636 will fix the missing line number.
`dis` shouldn't crash however, even if there is no line number.
Could you change the test to
```
def bug42562():
      pass

bug42562.__code__ = bug42562.__code__.replace(co_linetable=b'\x04\x80\xff\x80')
```
to make the test robust against changes like https://github.com/python/cpython/pull/23636

**Comment by uriyyo** [comment]
@markshannon I have updated test.

Should I also revert `show_lineno = bool(linestarts)`?

File: Lib/test/test_dis.py [hunk 1]
```diff
@@ -166,6 +166,19 @@ def bug1333982(x=[]):
        bug1333982.__code__.co_firstlineno + 2,
        bug1333982.__code__.co_firstlineno + 1)
 
+
+def bug42562():
+    pass
+
+
```
Comment (markshannon) [inline Lib/test/test_dis.py:173]: Better add a comment here, otherwise this is a bit mysterious.
`# Set line number for 'pass' to None`

**Review by markshannon — CHANGES_REQUESTED** [review]
Needs a comment, otherwise LGTM.

File: Lib/test/test_dis.py (hunk 1 above)
Comment (uriyyo) [reply Lib/test/test_dis.py:173]: My bad, it's always better to explain magic things)

**Comment by uriyyo** [comment]
I have made the requested changes; please review again

**Comment by uriyyo** [comment]
Should we update PR title and `News`?

**Comment by markshannon** [comment]
> Should we update PR title and `News`?

Yes. Something like "has no line numbers" instead of "has only annotations"

**Comment by uriyyo** [comment]
I have updated PR title and `News` section.

**Pull request merged**